### PR TITLE
M005, Tab Studies

### PR DESCRIPTION
**Pull request by RichardGarcia**
Add Tab studies - M005. I also tried using Paper that will be use in other Atom, Molecules or Organisms..

### CODE REVIEW

File: ui/components/atoms/A001-Button/index.js [hunk 1]
```diff
@@ -13,7 +14,7 @@ import schema from './schema'
 // Define Atom as JSX
 const A001 = props => {
   return (
-    <Button variant={'contained'} color={props.color}>
+    <Button variant={props.variant} color={props.color}>
```
Comment (agentlewis) [inline ui/components/atoms/A001-Button/index.js]: shouldn't this be contained?

File: ui/components/atoms/A001-Button/schema.js [hunk 2]
```diff
@@ -1,14 +1,40 @@
...
-  $schema: 'http://json-schema.org/schema#',
-  type: 'object',
-  properties: {
-    color: {
-      type: 'string'
-    },
-    buttonTitle: {
-      type: 'string'
-    }
+  $schema: "http://json-schema.org/schema#",
```
Comment (agentlewis) [inline ui/components/atoms/A001-Button/schema.js]: update to all single quotes

File: ui/components/atoms/A001-Button/stories.js [hunk 3]
```diff
@@ -6,12 +6,9 @@ import Button from '.'
 
 // List of stories of use cases for atom
 storiesOf('Atoms|A001 - Button', module)
-  .add('Default', () => (
-    <Button buttonTitle={'default button'} color={'default'} />
+  .add('Ok Button', () => (
+    <Button buttonTitle={'Ok'} color={'default'} variant={'contained'} />
```
Comment (agentlewis) [inline ui/components/atoms/A001-Button/stories.js]: variant should be hard coded in the atom

File: ui/components/atoms/A001-Button/stories.js [hunk 4]
```diff
@@ -6,12 +6,9 @@ import Button from '.'
...
   ))
-  .add('Primary', () => (
-    <Button buttonTitle={'primary button'} color={'primary'} />
-  ))
-  .add('Secondary', () => (
-    <Button buttonTitle={'secondary button'} color={'secondary'} />
+  .add('Participate Order Button', () => (
+    <Button buttonTitle={'Participate Order'} color={'default'} variant={'contained'} />
```
Comment (agentlewis) [inline ui/components/atoms/A001-Button/stories.js]: as above

File: ui/components/atoms/A005-Tab/index.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+// Import JSON schema
+import schema from './schema'
+
+// Import styles here - see example setup for styles at ui/example/components/atoms/checkbox
+
+// Define Atom as JSX
+const M005 = props => {
+  return (
+    <Tab icon={<Icon iconname={props.iconname}/>} label={props.label} />
```
Comment (agentlewis) [inline ui/components/atoms/A005-Tab/index.js]: how is this working - there is no tab import?

Comment (agentlewis) [reply ui/components/atoms/A005-Tab/index.js]: also icons should be imported from the particle no?

File: ui/components/atoms/A005-Tab/schema.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+      type: "string"
+    },
+    disabled: {
+      type: "boolean"
+    },
+    icon: {
+      type: "string"
+    }
+  },
+  required: []
```
Comment (agentlewis) [inline ui/components/atoms/A005-Tab/schema.js]: use oneOf instead:

    "oneOf": [
        {
            "required": [
                "copyAll"
            ]
        },
        {
            "required": [
                "fileNames"
            ]
        },
        {
            "anyOf": [
                {
                    "required": [
                        "matchesFiles"
                    ]
                },
                {
                    "required": [
                        "doesntMatchFiles"
                    ]
                }
            ]
        }
    ]

File: ui/components/atoms/A005-Tab/stories.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
+import React from 'react'
+import { storiesOf } from '@storybook/react'
+
+// Import index.js from the same folder
+import { Tab } from '@material-ui/core'
```
Comment (agentlewis) [inline ui/components/atoms/A005-Tab/stories.js]: look at the comment on line 4 - your not using the actual code your writing!

File: ui/components/molecules/M004-AppBar/index.js [hunk 8]
```diff
@@ -2,12 +2,8 @@
 import React from 'react'
 import createPropTypes from 'json-schema-prop-types'
 
-import { withStyles } from '@material-ui/core/styles';
-
 // Import from UI library
-import AppBar from '@material-ui/core/AppBar';
-import Toolbar from '@material-ui/core/Toolbar';
-import Typography from '@material-ui/core/Typography';
+import { AppBar, Toolbar, Typography } from '@material-ui/core';
```
Comment (agentlewis) [inline ui/components/molecules/M004-AppBar/index.js]: dont import from material in a molecule

File: ui/components/particles/P001-Icon/index.js [hunk 9]
```diff
@@ -16,7 +16,10 @@ const materialMap = {
   minus: 'minus',
   hamburger: 'menu',
```
Comment (agentlewis) [inline ui/components/particles/P001-Icon/index.js]: all of these maps should be using the material ui names

File: ui/components/particles/P001-Icon/schema.js [hunk 10]
```diff
@@ -0,0 +1,43 @@
+export default {
+  "$schema": "http://json-schema.org/schema#",
```
Comment (agentlewis) [inline ui/components/particles/P001-Icon/schema.js]: single quotes

File: ui/components/particles/P002-Typography/schema.js [hunk 11]
```diff
@@ -0,0 +1,41 @@
+export default {
+  "$schema": "http://json-schema.org/schema#",
```
Comment (agentlewis) [inline ui/components/particles/P002-Typography/schema.js]: single quotes

File: ui/components/molecules/M004-AppBar/index.js (hunk 8 above)
Comment (RichardGarcia) [reply ui/components/molecules/M004-AppBar/index.js]: sorry this is not part of the task/branch yet

File: ui/components/atoms/A005-Tab/index.js [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+// Import JSON schema
+import schema from './schema'
+
+// Import styles here - see example setup for styles at ui/example/components/atoms/checkbox
+import Icon from '../../particles/P001-Icon'
+
+// Define Atom as JSX
+const A005 = props => {
+  return (
+    <Tab icon={props.icon} />
```
Comment (agentlewis) [inline ui/components/atoms/A005-Tab/index.js]: My understanding was that we would be passing in an <Icon iconname=props.iconname/> here?

File: ui/components/atoms/A005-Tab/stories.js [hunk 13]
```diff
@@ -0,0 +1,16 @@
+import React from 'react'
+import { storiesOf } from '@storybook/react'
+
+// Import index.js from the same folder
+import Tab from '.'
+import Icon from '.'
```
Comment (agentlewis) [inline ui/components/atoms/A005-Tab/stories.js]: you shouldn't need to import the icon here - just tell the tab the iconname as per the schema

File: ui/components/particles/P001-Icon/index.js [hunk 14]
```diff
@@ -14,9 +14,12 @@ const materialMap = {
   add: 'add',
   cancel: 'cancel',
   minus: 'minus',
-  hamburger: 'menu',
+  menu: 'menu',
   gear: 'settings',
```
Comment (agentlewis) [inline ui/components/particles/P001-Icon/index.js]: update to settings: 'settings'

all of these should be like this.

File: ui/components/particles/P001-Icon/index.js [hunk 15]
```diff
@@ -14,9 +14,12 @@ const materialMap = {
...
-  question: 'help'
+  question: 'help',
```
Comment (agentlewis) [inline ui/components/particles/P001-Icon/index.js]: help: 'help

File: ui/components/particles/P001-Icon/schema.js (hunk 10 above)
Comment (agentlewis) [reply ui/components/particles/P001-Icon/schema.js]: single quotes

File: ui/components/atoms/A005-Tab/index.js (hunk 12 above)
Comment (agentlewis) [reply ui/components/atoms/A005-Tab/index.js]: String

Comment (RichardGarcia) [reply ui/components/atoms/A005-Tab/index.js]: Dan, I'm working on this right now

**Comment by agentlewis** [comment]
Done - great stuff. Now let’s focus on small branches going forward ;)

**Pull request merged**